src/index.ts

```diff
@@ -24,6 +25,45 @@ if (parseInt(process.version.split('.')[0]) < 16) {
   throw new Error(`In order to work, the package ${name} needs to run in Node v16 or newer to handle streams properly.`)
 }
 
+async function downloadProfileAvatars(
+  components: Pick<SnapshotsFetcherComponents, 'fetcher' | 'logs' | 'metrics' | 'storage'>,
+  logger: ILoggerComponent.ILogger,
+  presentInServers: string[],
+  _serverMapLRU: Map<Server, number>,
+  targetFolder: string,
+  maxRetries: number,
+  waitTimeBetweenRetries: number,
+  entityMetadata:
+    {
+      type: string
+      metadata?: any
+      content?: ContentMapping[] | undefined
+    }) {
+  const allAvatars: any[] = entityMetadata.metadata?.avatars ?? []
+  const snapshots = allAvatars.flatMap(avatar => Object.values(avatar.avatar.snapshots ?? {}) as string[])
+    .filter(snapshot => !!snapshot)
+    .map(snapshot => {
+      const matches = snapshot.match(/^http.*\/content\/contents\/(.*)/)
+      return matches ? matches[1] : snapshot
+    })
+    .filter(snapshot => !entityMetadata.content || entityMetadata.content.find(content => content.hash === snapshot) === undefined)
+  if (snapshots.length > 0) {
+    logger.info(`Downloading snapshots ${snapshots} for fixing entity ${JSON.stringify(entityMetadata)}`)
```

Comment:
```suggestion
    logger.info(`Downloading snapshots ${snapshots} for fixing profile entity ${JSON.stringify(entityMetadata)}`)
```

Note that these snapshots are not the same as the snapshots used for sync